### ISSUE TITLE
Support for compiling with c++ files

### ISSUE THREAD

**Issue by rwcarlsen** [karalabe/xgo#16]
Go now supports compiling c++ files (as long as you write a c interface over the c++ api of course).  It would be fantastic if cross platform c++ toolchains could be added to the docker image to enable doing this with xgo.  It shouldn't be too difficult, but I'm not enough of a docker expert to know where things need to change.  I think installing appropriate c++ compilers and adding the ``CXX=...`` for each build is about all that would be necessary.

**Comment by karalabe** [comment]
If you can put together a simple enough C++ dependent project and put it on github, I might play around with it if I have some time available :) I don't really have the time to start fiddling with cgo to ensure that that part is really correct and issues are/or are not in xgo :)

**Comment by rwcarlsen** [comment]
github.com/rwcarlsen/cyan/cmd/cyan binary imports the github.com/cyan/nuc
package which has some cgo with c++ stuff in it.

**Comment by rwcarlsen** [comment]
Sorry - that last link should be github.com/rwcarlsen/cyan/nuc.

**Comment by rwcarlsen** [comment]
I'll note that xgo currently works out of the box with c++ files for the linux-amd64 architecture.

**Comment by karalabe** [comment]
Hey :)

  Sorry it's been quite some time. I haven't had the time to work on xgo in the past months so couldn't properly address this issue. Your PR indeed solves C++ cross compilation for the main platforms, but I didn't want to merge it in without having a proper solution for all supported targets. Furthermore, I've also updated xgo to Go 1.5+, which uses different cross compilation semantics, as well as added support for android cross compilation, so your original PR got very out of date and also had understandably had merge conflicts.

  However, I figured that if I started patching up xgo for new features, I might as well address your request of being able to cross compile C++, so I've pushed https://github.com/karalabe/xgo/commit/e95f85ee814291bd9ec4e5451bad9518d607ce23 just now, which should make xgo happy on all platforms.

  Compiling your specific code sample:

```
$ xgo github.com/rwcarlsen/cyan/nuc
[...]

$ ls -al
-rw-r--r--  1 root     root      628738 Sep 16 17:03 nuc-android-21-arm
-rw-r--r--  1 root     root      350926 Sep 16 17:03 nuc-darwin-386
-rw-r--r--  1 root     root      353080 Sep 16 17:03 nuc-darwin-amd64
-rw-r--r--  1 root     root     1388814 Sep 16 17:03 nuc-linux-386
-rw-r--r--  1 root     root     2641598 Sep 16 17:03 nuc-linux-amd64
-rw-r--r--  1 root     root     1628688 Sep 16 17:03 nuc-linux-arm
-rw-r--r--  1 root     root     1953608 Sep 16 17:03 nuc-windows-386.exe
-rw-r--r--  1 root     root     1798090 Sep 16 17:03 nuc-windows-amd64.exe
```

Sorry for not being able to address this in a timely fashion :)

PS: Docker is still crunching through the updates, so it will probably take half an hour until the fresh images are out.
